pkg/config/expand.go

```diff
@@ -275,7 +275,7 @@ func (dc *DeploymentConfig) applyUseModules() error {
 
 				// this module contains information about the target module that
 				// was specified by the user in the blueprint
-				toMod, err := toGroup.getModuleByID(modRef.ID)
+				toMod, err := toGroup.getModuleByID(modRef.ToModuleID)
```

Comment:
nit: I would recommend using the getter now that you have it, here and elsewhere, and consider making the struct variable private.

Reply:
Out-of-band discussion resulted in a plan to use getters where necessary and unexported variables within the Go package.

Reply:
Discussed offline. Fields are now private. We can switch to using getters later if there is value.